apps/posts/models.py

```diff
@@ -59,7 +61,28 @@ class Post(models.Model):
                                 blank=True
                                 )
     date_created =  models.DateTimeField(auto_now_add=True)
-    #will figure out how to do the expiration thing here
 
     def __str__(self):
         return self.caption
+
+
+# a function that clears items that had been in the db 
+# for 2mins or more(for the purpose of testing)
+def clear_items(sender, instance, created, **kwargs):
+    if created:
+        items = Post.objects.all()
+
+        if items:
+            for item in items:
+                date_created = item.date_created
+                current_date = datetime.now(timezone.utc)
+                time_delta = current_date - date_created
+                minutes = time_delta.total_seconds() / 60
+
+                if minutes >= 2:
+                    item.delete()
```

Comment:
Use the reversed logic. Instead of putting everything inside `if created` do `if not created` and return an error. Same for `if items` (do `if not items` and alter the logic)

Reply:
noted!